src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.MethodSymbolKey.cs

```diff
@@ -85,10 +91,16 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? fa
                 using var result = PooledArrayBuilder<IMethodSymbol>.GetInstance();
                 foreach (var method in constructedFrom.OfType<IMethodSymbol>())
                 {
-                    if (method.TypeParameters.Length == typeArgumentArray.Length)
-                    {
-                        result.AddIfNotNull(method.Construct(typeArgumentArray));
-                    }
+                    if (!method.Equals(method.ConstructedFrom))
+                        continue;
+
+                    if (method.Arity == 0)
+                        continue;
```

Comment:
these two checks exactly match the condition checked for here: https://github.com/dotnet/roslyn/blob/4df988c1e87760c53df3448fbe6bbb34795360d2/src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs#L845

just making sure we don't violate this ourselves.